src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -2381,47 +2853,73 @@ RegularTable(Oid relationId)
 
 
 /*
- * CopyLocalDataIntoShards copies data from the local table, which is hidden
- * after converting it to a distributed table, into the shards of the distributed
- * table. For partitioned tables, this functions returns without copying the data
+ * CopyLocalDataIntoShards is a wrapper around CopyFromLocalTableIntoDistTable
+ * to copy data from the local table, which is hidden after converting it to a
+ * distributed table, into the shards of the distributed table.
+ *
+ * After copying local data into the distributed table, the local data remains
+ * in place and should be truncated at a later time.
+ */
+static void
+CopyLocalDataIntoShards(Oid distributedTableId)
+{
+	bool skipCopyToLocalPlacements = false;
+	uint64 rowsCopied = CopyFromLocalTableIntoDistTable(distributedTableId,
+														distributedTableId,
+														skipCopyToLocalPlacements);
+	if (rowsCopied > 0)
+	{
+		char *qualifiedRelationName =
+			generate_qualified_relation_name(distributedTableId);
+		ereport(NOTICE, (errmsg("copying the data has completed"),
+						 errdetail("The local data in the table is no longer visible, "
+								   "but is still on disk."),
+						 errhint("To remove the local data, run: SELECT "
+								 "truncate_local_data_after_distributing_table($$%s$$)",
+								 qualifiedRelationName)));
+	}
+}
+
+
+/*
+ * CopyFromLocalTableIntoDistTable copies data from given local table into
+ * the shards of given distributed table.
+ *
+ * For partitioned tables, this functions returns without copying the data
  * because we call this function for both partitioned tables and its partitions.
  * Returning early saves us from copying data to workers twice.
  *
  * This function uses CitusCopyDestReceiver to invoke the distributed COPY logic.
- * We cannot use a regular COPY here since that cannot read from a table. Instead
- * we read from the table and pass each tuple to the CitusCopyDestReceiver which
+ * We cannot use a regular COPY here since localTableId might indeed be a Citus
+ * table too. In that case, Citus would intercept queries on this table in the
+ * planner and executor hooks but we assume that caller wants to read the local
+ * data in the shell table. To keep it simple, we perform a heap scan directly
+ * on the table.
```

Comment:
more accurately, we want to read from the local table and we cannot use SELECT because it would get intercepted and use the still-empty placements